packages/components/src/components/SquiggleViewer/ViewerProvider.tsx

```diff
@@ -334,7 +334,14 @@ export const ViewerProvider: FC<
           const { calculator, path } = action.payload;
           setLocalItemState(path, (state) => ({
             ...state,
-            calculator: calculator,
+            calculator:
+              state.calculator?.hashString === calculator.hashString
+                ? {
+                    // merge with existing value
```

Comment:
What's the reasoning for this? (Maybe leave in comment on code)

Reply:
Since both Calculator and CalculatorResult send stuff to ViewerContext, `CALCULATOR_UPDATE` action payload now doesn't have to include all fields. `<Calculator>` sends `formInputs` and `inputValues`, while `<CalculatorResult>` sends `fnValue`.